app/controllers/actions/DataRequiredAction.scala

```diff
@@ -35,7 +34,13 @@ class DataRequiredActionImpl @Inject() (implicit val executionContext: Execution
     request: OptionalDataRequest[A]
   ): Future[Either[Result, DataRequest[A]]] = {
     val result = request.userAnswers match {
-      case None       => Left(Redirect(routes.JourneyRecoveryController.onPageLoad()))
+
+      case None       =>
+        if (request.toString.contains("/cancel")) {
+          Left(Redirect(routes.YourApplicationHasBeenCancelledController.onPageLoad()))
```

Comment:
Can we add a test for DataRequiredAction?

Reply:
Yeah this should have one really 👍